### ISSUE TITLE
Add catkin_pkg to requirements in setup.py

### ISSUE THREAD

**Issue by fedeceola** [ethz-asl/vgn#29]
Hi, I would like to install VGN from the setup.py in my project. Would it be possible to add the installation of `catkin_pkg` as a requirement in the VGN's setup.py? Thanks in advance!

**Comment by mbreyer** [comment]
Hi, I typically first installed the requirements from `requirements.txt` which contains `catkin_pkg` before running `pip install`. I guess one could alternatively add the dependencies to the `install_requires` field of the `setup` method inside setup.py, but I have never tried it out since the [ROS documentation](http://docs.ros.org/en/jade/api/catkin/html/user_guide/setup_dot_py.html) states that this field has no meaning in ROS and I wanted to keep it compatible with catkin as well.

**Comment by fedeceola** [comment]
Hi, thanks for your reply. I'm currently installing the requirements before running `pip install` as well. What I would like to do is exactly what you mentioned, but adding the `catkin_pkg` in the `install_requires` field of my `setup.py` doesn't work. As far as I know it's not possible to decide the order of installation of the packages. Therefore, the installation of `vgn` fails because `vgn`'s `setup.py` imports `catkin_pkg` before installing it.

**Comment by mbreyer** [comment]
That makes sense. If you do not install the package with catkin anyways, a solution to your problem could be to simply remove the catkin_pkg dependency and add the remaining dependencies to setup.py.

**Comment by fedeceola** [comment]
Yes, but in that case the problem is again the installation of `vgn`. I think that the only way to solve the problem is to remove the `catkin_pkg` dependency from the `vgn`'s `setup.py`. If you think that this is doable, to me it would be helpful, otherwise installing `catkin_pkg` before `vgn` as I have done so far is not a big problem.

**Comment by mbreyer** [comment]
That should work. Can you try replacing the existing setup.py with the following code?

```python
from setuptools import setup

setup(
    name="vgn",
    version="0.1.0",
    author="Michel Breyer",
    author_email="mbreyer@ethz.ch",
    description="Volumetric Grasping Network.",
    packages=[
        "vgn",
        "vgn.utils",
    ],
    package_dir={"": "src"},
    install_requires=[
        "numpy",
        "scipy",
        "pandas",
        "matplotlib",
        "mpi4py",
        "open3d",
        "pybullet==2.7.9",
        "torch",
        "pytorch-ignite",
        "tensorboard",
        "tqdm",
    ],
)

```

**Comment by fedeceola** [comment]
Yes, this `setup.py` works!

**Comment by mbreyer** [comment]
Awesome!
